src/message.rs

```diff
@@ -1369,6 +1369,58 @@ pub fn get_deaddrop_msg_cnt(context: &Context) -> usize {
     }
 }
 
+pub fn estimate_deletion_cnt(
+    context: &Context,
+    from_server: bool,
+    seconds: i64,
+) -> Result<usize, Error> {
+    let self_chat_id = chat::lookup_by_contact_id(context, DC_CONTACT_ID_SELF)
+        .unwrap_or_default()
+        .0;
+    let threshold_timestamp = time() - seconds;
+
+    let cnt: isize;
+    if from_server {
+        cnt = context.sql.query_row(
+            "SELECT COUNT(*)
+             FROM msgs m
+             WHERE m.id > ?
+               AND (state = ? OR state >= ?)
+               AND chat_id != ?
+               AND timestamp < ?
+               AND server_uid != 0;",
+            params![
+                DC_MSG_ID_LAST_SPECIAL,
+                MessageState::InSeen,
+                MessageState::OutFailed,
+                self_chat_id,
+                threshold_timestamp
+            ],
+            |row| row.get(0),
+        )?;
+    } else {
+        cnt = context.sql.query_row(
+            "SELECT COUNT(*)
+             FROM msgs m
+             WHERE m.id > ?
+               AND (state = ? OR state >= ?)
+               AND chat_id != ?
+               AND timestamp < ?
+               AND chat_id != ?;",
```

Comment:
In addition to trash chat, need to check then `hidden` flag. MDNs and location.kml messages are stored as hidden messages in non-trash chat.

Reply:
you're right.